code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -8,71 +8,250 @@ var/bomb_set
 	use_power = 0
 	var/previous_level = ""
 	var/lastentered = ""
 	var/immobile = 0 //Not all nukes should be moved
+	var/obj/item/nuke_core/core = null
+	var/construction_state = 6
 
 /obj/machinery/nuclearbomb/selfdestruct
 	name = "station self-destruct terminal"
 	desc = "For when it all gets too much to bear. Do not taunt."
 	icon = 'icons/obj/machines/bignuke.dmi'
+	icon_state = "nuclearbomb_base"
 	anchored = 1 //stops it being moved
 	immobile = 1 //prevents it from ever being moved
 	layer = 4
+	var/icon/lights = null
+	var/icon/panel = null
+	var/icon/interior = null
+	var/icon/glow = null
+
+/obj/machinery/nuclearbomb/selfdestruct/New()
+	core = new /obj/item/nuke_core(src)
+	lights = new /icon(icon,"lights-off")
+	panel = new /icon(icon,"panel-overlay")
+	overlays += lights
+	overlays += panel
+	..()
+
+/obj/machinery/nuclearbomb/selfdestruct/attackby(obj/item/I, mob/user, params)
+	switch(construction_state)
+		if(6)
+			if(istype(I, /obj/item/weapon/screwdriver/nuke))
+				playsound(loc, 'sound/items/Screwdriver.ogg', 100, 1)
+				user << "<span class='notice'>You start removing the front panel's screws...</span>"
+				if(do_after(user, 100))
+					construction_state = 5
+					user << "<span class='notice'>You remove the screws and the front panel slides open.</span>"
+					update_icon()
+				return
+		if(4,5)
+			if((construction_state == 5) && istype(I, /obj/item/weapon/wirecutters))
+				playsound(loc, 'sound/effects/sparks4.ogg', 100, 1)
+				playsound(loc, 'sound/effects/EMPulse.ogg', 100, 1)
+				user << "<span class='warning'>You must have cut the wrong wire!</span>"
+				for(var/mob/living/L in range(5,src))
+					L.irradiate(200)
+				construction_state = 4 //cant cut wires no more
+			else
+				if(istype(I, /obj/item/stack/cable_coil))
+					var/obj/item/stack/cable_coil/S = I
+					user << "<span class='notice'>You start tying the uncountable wires...</span>"
+					if(do_after(user,30))
+						if(S.use(15))
+							user << "<span class='notice'>You tie the uncountable wires with some cable, clearing the insides of [src].</span>"
+							construction_state = 3
+							update_icon()
+						else
+							user << "<span class='warning'>You need more cable to do that.</span>"
+					return
+				if(istype(I, /obj/item/weapon/restraints/handcuffs/cable))
+					user << "<span class='notice'>You start tying the uncountable wires...</span>"
+					if(do_after(user,30))
+						if(user.drop_item())
+							qdel(I)
+							user << "<span class='notice'>You tie the uncountable wires with the [I], clearing the insides of [src].</span>"
+							construction_state = 3
+							update_icon()
+					return
+		if(3)
+			if(istype(I, /obj/item/weapon/pen))
+				user << "<span class='notice'>You start drawing cut lines...</span>"
+				if(do_after(user,30))
+					user << "<span class='notice'>You draw cut lines inside [src].</span>"
+					construction_state = 2
+					update_icon()
+				return
+			if(istype(I, /obj/item/toy/crayon))

```

Comment:
this should really just be a pen, everybody gets a pen in their PDA so it's not like they're rare.